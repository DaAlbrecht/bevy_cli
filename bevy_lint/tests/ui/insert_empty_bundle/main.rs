#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_empty_bundle)]

use bevy::prelude::*;
use std::f32::consts::PI;

fn main() {
    App::new().add_systems(Startup, test);
}

fn test(mut commands: Commands) {
    commands.spawn((
        Name::new("Decal"),
        Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
        //~^ ERROR: Expression returns `()` and results in an empty bundle being inserted
    ));
}
