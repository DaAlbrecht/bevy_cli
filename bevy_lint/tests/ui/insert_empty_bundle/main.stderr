error: Expression returns `()` and results in an empty bundle being inserted
  --> tests/ui/insert_empty_bundle/main.rs:15:64
   |
15 |         Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
   |                                                                ^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/insert_empty_bundle/main.rs:3:9
   |
3  | #![deny(bevy::insert_empty_bundle)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

